oak_attestation_verification/src/verifier.rs

```diff
@@ -660,23 +673,42 @@ fn verify_measurement_digest(
                 &public_keys.endorser_public_key,
                 &public_keys.rekor_public_key,
             )?;
-            let expected = get_digest(&parse_endorsement_statement(&endorsement.endorsement)?)?;
-            verify_hex_digests(&actual, &expected)
-        }
-        Some(binary_reference_value::Type::Digests(expected_digests)) => {
-            if expected_digests.digests.iter().any(|expected_digest| {
-                let expected = raw_to_hex_digest(expected_digest);
-                verify_hex_digests(&actual, &expected).is_ok()
-            }) {
-                Ok(())
-            } else {
-                Err(anyhow::anyhow!("measurement digest does not match any reference values"))
-            }
+            Ok(ExpectedDigests {
+                r#type: Some(expected_digests::Type::Digests(HexDigests {
+                    digests: vec![get_digest(&parse_endorsement_statement(
+                        &endorsement.endorsement,
+                    )?)?],
+                })),
+            })
         }
+        Some(binary_reference_value::Type::Digests(expected_digests)) => Ok(ExpectedDigests {
+            r#type: Some(expected_digests::Type::Digests(HexDigests {
+                digests: expected_digests.digests.iter().map(raw_to_hex_digest).collect(),
+            })),
+        }),
         None => Err(anyhow::anyhow!("empty binary reference value")),
     }
 }
 
+/// Verifies the measurement digest value against a reference value and
+/// the expected digests calculated from endorsements and reference values.
+fn compare_measurement_digest(
+    measurement: &RawDigest,
+    expected: &ExpectedDigests,
+) -> anyhow::Result<()> {
+    let actual = raw_to_hex_digest(measurement);
```

Comment:
Side question: It seems there's a lot of conversion to hex digest for comparison. 

I noticed there's a `is_raw_digest_match` comparison helper, but it first converts to hex, and then uses the hex comparison function. 

Would it make sense to work directly with raw digests to avoid some of the conversion overhead?

Reply:
cc @thmsbinder

Reply:
If we are going to keep using hex digests, I would like to at least reverse this to always prefer raw digests. The evidence is always a raw digest, so I would expect that the exptected values should also contain raw digests and the comparison should directly compare the two.